### PR TITLE
5 hotfix-parser

### PR DESCRIPTION
**Pull request by ghost**
hot fix branch for the parser
changes requested by @Sebastian-Fransson & @pontuslaestadius 
Renamed classes and varaibles for redability
more comments
minor fixes

### CODE REVIEW

File: Application/src/model/sequenceDiagramParser/Processes.java [hunk 1]
```diff
@@ -3,9 +3,9 @@
 import com.google.gson.annotations.SerializedName;
 
 /**
- * This Class is part Sequence Diagram Object for the parser from JSON to POJO (Plain Old Java Objects)
+ * This Class is part of the Sequence Diagram Object for the parser from JSON to POJO (Plain Old Java Objects)
  * The getters and setters and their declarations are made to adhere with a predetermined JSON format
- *
+ * This Poetesses class will contain the class names in the parsed sequence diagram
```
Comment (kosara-g) [inline Application/src/model/sequenceDiagramParser/Processes.java]: poetic

**Review by kosara-g — APPROVED** [review]
Seems good.

File: Application/src/model/sequenceDiagramParser/Processes.java (hunk 1 above)
Comment (ghost) [reply Application/src/model/sequenceDiagramParser/Processes.java]: 😆

File: Application/src/model/Parser.java [hunk 2]
```diff
@@ -73,43 +75,48 @@ public void parseSequenceDiagram(String inputJSON) {
...
+            // Creates an instance of the diagram's meta
+            model.sequenceDiagramParser.Meta meta = parsedSequenceDiagram.getMeta();
+            // Assigns a String to the diagram's format
+            String format = meta.getFormat();
+            // Assigns a String to the diagram's extensions
+            List<Object> extensions = meta.getExtensions();
+            // Assigns a String to the diagram's version
+            String version = meta.getVersion();
+            // Assigns a String to the type of the diagram
+            String type = parsedSequenceDiagram.getType();
```
Comment (pontuslaestadius) [inline Application/src/model/Parser.java:90]: All of these variables are unused?

File: Application/src/model/Parser.java [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
+import java.util.List;
+
 
 /**
  * @author Rashad Kamsheh & Isabelle Törnqvist
- * @version 1.1
+ * @version 1.2
  * @since 2017-10-16
- *
+ * <p>
```
Comment (pontuslaestadius) [inline Application/src/model/Parser.java]: why are you opening p tags but not closing them?

File: Application/src/model/Parser.java [hunk 4]
```diff
@@ -73,43 +75,48 @@ public void parseSequenceDiagram(String inputJSON) {
...
             // get the elements of the Parallel Diagram
-            for (ContentArray parallelDiagramElement : parsedDiagram.getDiagram().getContent().get(1).getContent()) {
+            for (Messages parallelDiagramElement : parsedSequenceDiagram.getDiagram().getContent().get(1).getMessages()) {
                 tempParallelDiagram += "{" + parallelDiagramElement.getFrom() + "," + parallelDiagramElement.getTo() + ",[" + parallelDiagramElement.getMessage().get(0) + "]" + "}" + ",";
                 //removing extra comma
                 parallelDiagramString = (tempParallelDiagram.substring(0, tempParallelDiagram.length() - 1));
             }
 
-        } catch (NullPointerException e) {
+        } catch (Exception e) {
```
Comment (pontuslaestadius) [inline Application/src/model/Parser.java:118]: All errors are syntax error related?
Also the syntax error says 'import failed' not a 'parsing failed'?

**Review by pontuslaestadius — CHANGES_REQUESTED** [review]
You still have about 20 unused getters/setters? Isn't it still hard to understand if it's a lot code that does nothing?

File: Application/src/model/Parser.java (hunk 4 above)
Comment (ghost) [reply Application/src/model/Parser.java:118]: Not all users know what parsing is, so let's keep it at "import failed"
are there any other parsing error not related to a bad syntax/ unrecognizable syntax?

File: Application/src/model/Parser.java (hunk 2 above)
Comment (ghost) [reply Application/src/model/Parser.java:90]: for now

**Review by pontuslaestadius — APPROVED** [review]
I still don't know whats going on.

**Pull request merged**